commit 0dde2440a77fe48495c8e99108003291db81e1a5
Author: Avraham Stern <avraham.stern@intel.com>
Date:   Wed Jul 11 10:30:20 2018 +0300

    iwlwifi: mvm: toggle tx antenna if tx fails during connection establishment
    
    If tx fails during connection establishment, try another antenna for
    the next tx. This will increase the chance to establish connection if
    one of the antennas is blocked.  Note that the antenna is toggled even
    when failing to tx data frames since connection establishment may use
    EAPOLs for 802.1X authentication/ 4 way handshake.
    
    Signed-off-by: Avraham Stern <avraham.stern@intel.com>
    Signed-off-by: Luca Coelho <luciano.coelho@intel.com>

diff --git a/drivers/net/wireless/intel/iwlwifi/mvm/sta.c b/drivers/net/wireless/intel/iwlwifi/mvm/sta.c
index 79ccc698ad83..e82194836f03 100644
--- a/drivers/net/wireless/intel/iwlwifi/mvm/sta.c
+++ b/drivers/net/wireless/intel/iwlwifi/mvm/sta.c
@@ -1768,6 +1768,8 @@ int iwl_mvm_add_sta(struct iwl_mvm *mvm,
 	if (iwl_mvm_has_tlc_offload(mvm))
 		iwl_mvm_rs_add_sta(mvm, mvm_sta);
 
+	iwl_mvm_toggle_tx_ant(mvm, &mvm_sta->tx_ant);
+
 update_fw:
 	ret = iwl_mvm_sta_send_to_fw(mvm, sta, sta_update, sta_flags);
 	if (ret)